app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationController.scala

```diff
@@ -102,13 +102,47 @@ class BusinessIdentificationController @Inject()(
             if (validatedBusinessType.businessType == Invalid)
               Redirect(routes.BusinessIdentificationController.showInvalidBusinessType)
             else
-              Redirect(
-                routes.BusinessIdentificationController.showBusinessDetailsForm(validatedBusinessType.businessType))
+              Redirect(routes.BusinessIdentificationController.showUtrForm(validatedBusinessType.businessType))
           }
         )
     }
   }
 
+  def showUtrForm(businessType: IdentifyBusinessType): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
passing `businessType` across endpoints back and forth is bas  - instead we should store it in the session and use it every where we need  -  but thats how existing code is written - needs to refactor the cache handling

Reply:
Sounds good I would agree storing it the session would be better.